game-core/src/test/java/games/strategy/triplea/util/MovableUnitsFilterTest.java

```diff
@@ -0,0 +1,115 @@
+package games.strategy.triplea.util;
+
+import static games.strategy.triplea.delegate.GameDataTestUtil.armour;
+import static games.strategy.triplea.delegate.GameDataTestUtil.germans;
+import static games.strategy.triplea.delegate.GameDataTestUtil.infantry;
+import static games.strategy.triplea.delegate.GameDataTestUtil.territory;
+import static games.strategy.triplea.delegate.MockDelegateBridge.advanceToStep;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.IsNull.nullValue;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerId;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.AbstractMoveDelegate;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.MockDelegateBridge;
+import games.strategy.triplea.xml.TestMapGameData;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.triplea.java.collections.CollectionUtils;
+
+public class MovableUnitsFilterTest {
+  final GameData gameData = TestMapGameData.REVISED.getGameData();
+  final PlayerId germans = germans(gameData);
+  final Territory germany = territory("Germany", gameData);
+  final Territory easternEurope = territory("Eastern Europe", gameData);
+  final Territory kareliaSsr = territory("Karelia S.S.R.", gameData);
+  final UnitType infantryType = infantry(gameData);
+  final UnitType armourType = armour(gameData);
+
+  public MovableUnitsFilterTest() throws Exception {}
```

Comment:
nit, empty constructor not needed in junit5 tests

Reply:
It's needed when it throws exception - which is needed because TestMapGameData.REVISED.getGameData() can throw.

Perhaps we should make getGameData() throw a RuntimeException so it's not needed?

@DanVanAtta WDYT?

Reply:
I sent a separate PR to change getGameData() that would allow this to be removed.

Reply:
I was wrong about the empty constructor not being needed here, I think Roi had the same comment on another PR.